benchmarks/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -285,22 +291,26 @@ int main_(std::vector<std::string> const &args, const MPI_Comm comm)
               << "#queries/MPI process    : " << n_queries << '\n'
               << "size of shift           : " << shift << '\n'
               << "dimension               : " << partition_dim << '\n'
+              << "shift-queries           : " << shift_queries << '\n'
               << '\n';
   }
 
   Kokkos::View<ArborX::Point *, DeviceType> random_points("random_points", 0);
+  Kokkos::View<ArborX::Point *, DeviceType> random_queries("random_queries", 0);
   {
-    // Random points are "reused" between building the tree and performing
-    // queries. Note that this means that for the points in the middle of
-    // the local domains there won't be any communication.
-    auto n = std::max(n_values, n_queries);
-    Kokkos::resize(random_points, n);
+    // By default, random points are "reused" between building the tree and
+    // performing queries. Note that this means that for the points in the
+    // middle of the local domains there won't be any communication.
+    Kokkos::resize(random_points, n_values);
+    Kokkos::resize(random_queries, n_queries);
```

Comment:
Can you elaborate? Assignment performs a shallow copy and we only use `n_queries` for the construction. Hence, we should not access uninitialized elements or get a segmentation fault.

Reply:
The problem is if you run `--values=100 --queries=10000` it will instead run `--values=100 --queries=100` due to the last assignment, as `random_points` size is 100.

Reply:
Correct, let me revise that.